dotnet-client-libraries/ITextMergingPDFs/ITextMergingPDFs/Merger.cs

```diff
@@ -0,0 +1,97 @@
+﻿using iText.Kernel.Pdf;
+using iText.Kernel.Utils;
+using iText.Layout;
+
+namespace ITextMergingPDFs
+{
+    public class Merger
+    {
+        private readonly string _path;
+
+        private enum CopyPages
+        {
+            AllPages,
+            FirstPage,
+            OddPages,
+            EvenPages
+        }
+
+        public static Merger Create(string path)
+        {
+            return new Merger(path);
+        }
+
+        private Merger(string path)
+        {
+            _path = path;
+        }
+
+        public string MergePDFs(string[] pdfFiles, string mergedPdfFileName) =>
+            DoMerging(pdfFiles, mergedPdfFileName, CopyPages.AllPages);
+
+        public string MergePDFsFirstPage(string[] pdfFiles, string mergedPdfFileName) =>
+            DoMerging(pdfFiles, mergedPdfFileName, CopyPages.FirstPage);
+
+        public string[] SplitDocument(string pdfFile)
+        {
+            var oddPages = DoMerging(new string[] { pdfFile }, GetFullFileName("odd.pdf"), CopyPages.OddPages);
+            var evenPages = DoMerging(new string[] { pdfFile }, GetFullFileName("even.pdf"), CopyPages.EvenPages);
+
+            return new string[] { oddPages, evenPages };
+        }
+
+        private string GetFullFileName(string mergedPdfFileName)
+        {
+            return System.IO.Path.Combine(_path, mergedPdfFileName);
+        }
+
+        private static string[] CheckParameters(string[] pdfFiles, string mergedPdfFileName)
+        {
+            if (string.IsNullOrWhiteSpace(mergedPdfFileName))
+                throw new ArgumentOutOfRangeException(nameof(mergedPdfFileName));
+
+            var existingFiles = pdfFiles.Where(File.Exists).ToArray();
+            if (existingFiles.Length == 0)
+                throw new ArgumentOutOfRangeException(nameof(pdfFiles));
+
+            return existingFiles;
+        }
+
+        private string DoMerging(string[] pdfFiles, string mergedPdfFileName, CopyPages copyPages)
+        {
+            var existingFiles = CheckParameters(pdfFiles, mergedPdfFileName);

```

Comment:
@MPrtenjak - with the suggested change above, this line can change to just `CheckParameters` or be renamed to `ValidateParameters` if you feel it would be more clear since it will either pass or throw.

Reply:
Repaired